tools/kicad_rs/src/parser.rs

```diff
@@ -243,21 +226,87 @@ fn get_component_attr_mapped(
         .flatten()
 }
 
-// str_if_nonempty trims the string reference s, and returns it as owned in an Option if non-empty.
-// As a special case, "~" also counts as "empty".
-fn str_if_nonempty(s: &str) -> Option<String> {
-    let s = s.trim();
-    if s.is_empty() || s == "~" {
-        None
-    } else {
-        Some(String::from(s))
+// IsTrueLike provides an is_true_like that turns a type into a bool given type-specific heuristic
+// rules.
+trait IsTrueLike {
+    fn is_true_like(&self) -> bool;
+}
+
+impl<T: AsRef<str>> IsTrueLike for T {
+    // is_true_like returns true is s is a "true-like" string like "true" or "1", otherwise false
+    fn is_true_like(&self) -> bool {
+        self.as_ref() == "true" || self.as_ref() == "1"
+    }
+}
+
+// OrEmptyStr provides an or_empty_str method that unwraps a string option in a safe way,
+// defaulting to an empty string if None. The inner string is cloned.
+trait OrEmptyStr {
+    // or_empty_str unwraps the string-like option such that if opt is None, a new, empty String is returned
+    fn or_empty_str(&self) -> String;
+}
+
+impl OrEmptyStr for Option<String> {
+    fn or_empty_str(&self) -> String {
+        self.clone().unwrap_or_else(|| String::new())
+    }
+}
+
+impl OrEmptyStr for Option<&str> {
+    fn or_empty_str(&self) -> String {
+        self.unwrap_or("").to_owned()
     }
 }
 
-// str_if_nonempty_opt passes the string through str_if_nonempty if the option is Some. In other words,
-// this function maps Some("") and Some("~") -> None, and lets all other values be.
-fn str_if_nonempty_opt(s_opt: Option<String>) -> Option<String> {
-    s_opt.map(|s| str_if_nonempty(&s)).flatten()
+// EmptyFilter provides a filter_empty method that filters a string or string option into a
+// string option, making sure "empty-like" strings make the option become None.
+trait EmptyFilter {
+    fn filter_empty(&self) -> Option<String>;
```

Comment:
Should this pass `String` ownership to the returned `Option<String>` instead of creating a clone?

Reply:
That would be ideal, but can't be done due to that we need to do `s.trim()` which gives back a `&str`. Theoretically maybe it'd be possible to take ownership of the owning string being trimmed, for just a subset of the source, but I guess that is an unsafe operation.